src/validation/3.0/invalid-property-value.rule.ts

```diff
@@ -138,6 +145,101 @@ export class Oas30InvalidPropertyValueValidationRule extends Oas30PathValidation
         return method === "put" || method === "post" || method === "options" || method === "patch";
     }
 
+    /**
+     * Returns resolved parameter definition or the address of the reference if parameter can not be resolved
+     * (e.g. when the pointer is to a non-existent value within the components).
+     * @param node - Parameter definition within a path item or operation
+     * @return {Oas30Parameter | string} - Returns (recursively) resolved parameter or a string with invalid ref.
+     */
+    private resolveParameter(node: Oas30Parameter): Oas30Parameter | string {
+        let resolvedParam = node;
+        while (resolvedParam.$ref !== undefined) {
+            const referencedNode = <Oas30Parameter>ReferenceUtil.resolveRef(resolvedParam.$ref, node.ownerDocument());
+            if (referencedNode === undefined || referencedNode === null) {
+                return resolvedParam.$ref;
+            }
+            resolvedParam = referencedNode;
+        }
+        return resolvedParam
+    }
+
+    /**
+     * Validates that all parameter name and "in" combinations are unique
+     * @param {Oas30Parameter[]} - List of parameters to check
+     * @return {boolean} - true if unique, false otherwise
+     */
+    private ensureUnique(params: Oas30Parameter[]) {
```

Comment:
Needs a return value in the method signature.

Reply:
Done.